packages/flutter_tools/lib/src/ios/ios_deploy.dart

```diff
@@ -235,6 +235,27 @@ class IOSDeploy {
     return true;
   }
 
+  Future<bool> isDeveloperModeEnabled({
+    required String deviceId,
+  }) async {
+    final List<String> launchCommand = <String>[
+      _binaryPath,
+      '--id',
+      deviceId,
+      '--check-developer-mode'
+    ];
+    final RunResult result = await _processUtils.run(
+      launchCommand,
+      environment: iosDeployEnv,
+    );
+    // Device does not have developer mode enabled.
+    if (result.stdout.trim().split('\n')[2] == 'Developer mode is not enabled.') {
```

Comment:
Note that Im splitting the stdout because this flag returns the same exitCode regardless if it succeeds or fails

Reply:
This would be extremely brittle to Apple changing this output (also you would use `LineSplitter`).

Instead `--json` and parse the `IsEnabled` output, see my example in https://github.com/flutter/flutter/issues/111988#issuecomment-1252994129


This also needs to handle devices running lower than iOS 16 that don't know anything about developer mode so there's nothing to turn on.  See the output in the same comment.